pkg/scheduler/framework/session_plugins.go

```diff
@@ -232,11 +232,39 @@ func (ssn *Session) QueueOrderFn(lQ, rQ *queue_info.QueueInfo, lJob, rJob *podgr
 		}
 	}
 
-	// If no queue order funcs, order queue by CreationTimestamp first, then by UID.
-	if lQ.CreationTimestamp.Equal(&rQ.CreationTimestamp) {
-		return lQ.UID < rQ.UID
+	// Queue order not resolved, fallback to job CreationTimestamp first, then by UID.
+	lJ, rJ := ssn.getJobsForQueueOrder(lJob, rJob, lVictims, rVictims)
+	if lJ != nil && rJ != nil {
+		if !lJ.CreationTimestamp.Equal(&rJ.CreationTimestamp) {
+			lJ.CreationTimestamp.Before(&rJ.CreationTimestamp)
+		}
+		return lJ.UID < rJ.UID
+	}
+
+	// Queue CreationTimestamp and UID as final tie-breaker
+	if !lQ.CreationTimestamp.Equal(&rQ.CreationTimestamp) {
```

Comment:
Is it possible to get to this line? I think that the only possible way is if one of the jobs is `nil` but then this is a weird situation to compare queues.

Reply:
I agree, but the code allows that. Maybe some further refactoring is needed here.
When this function received only two jobs as parameters it was much cleaner.